backend/fms_core/template_importer/importers/_generic.py

```diff
@@ -46,14 +57,30 @@ def import_template(self, file, format, dry_run):
                         reversion.set_comment("Template import - dry run")
                     transaction.set_rollback(True)
                 else:
-                    self.import_template_inner()
+                    # Save template on disk with modified name (append timestamp and id) and insert path in imported_file
+                    if user is not None:
+                        os.environ["TZ"] = settings.LOCAL_TZ
+                        time.tzset()
+                        new_file_name = f"{file_name}_{time.strftime('%Y-%m-%d_%H:%M:%S')}_{user.username}{file_format}"
+                        file_path = os.path.join(UPLOAD_PATH, new_file_name)
+                        try:
+                            self.imported_file = ImportedFile.objects.create(filename=new_file_name, location=file_path, created_by_id=user.id)
+                        except Exception as err:
+                            self.base_errors.append(err)
+                    self.import_template_inner()                            
                     reversion.set_comment("Template import")
 
             # Add processed rows with errors/warnings/diffs to self.previews_info list
             for sheet in list(self.sheets.values()):
                 preview_info = sheet.generate_preview_info_from_rows_results(rows_results=sheet.rows_results)
                 self.previews_info.append(preview_info)
 
+            # Save the template on the server if the template is valid.
```

Comment:
I wonder if this should be wrapped in a try block. What kind of response will the user get if an exception occurs? Has the user's data been commited to the DB, or will it roll back if an exception is raised?

Reply:
Lets say we do not have enough disk space to receive the file ? I think it will return a 500 internal server error. I think it will force a rollback of the whole request. I think we work with a system that wrap each request in a transaction. I am not sure though i will do some testing (writing to the same file name and see what happens). Having a try block will likely not prevent the issue but at least we can inform the user on the issue (the default message is likely very unhelpful).